File: exercicios/R/1116/1116.r
input = file('stdin','r')

n = as.integer(readLines(input,n=1))

for(i in 1:n){
    line = strsplit(readLines(input,n=1)," ")

    x = as.integer(line[[1]][1])
    y = as.integer(line[[1]][2])

    if(y == 0 ){
        write("divisao impossivel",'')
    }else{
        div = as.double(x/y)
        write(sprintf("%.1f",div),'')
    }
}

File: exercicios/JavaScript/1116/1116.js
var input = require('fs').readFileSync('stdin','utf8');
var lines = input.split('\n');

var n = parseInt(lines.shift());

for(let i=0; i<n; i++){
    var line1 =  lines.shift().split(' ');

    var x = parseInt(line1[0]);
    var y = parseInt(line1[1]);

    if(y == 0){
        console.log("divisao impossivel");
    }else{
        console.log(parseFloat(x/y).toFixed(1));
    }
}